sdk/include/debug.hh

```diff
@@ -352,7 +315,19 @@ struct DebugFormatArgumentAdaptor<CHERI::PermissionSet>
 	construct(CHERI::PermissionSet value)
 	{
 		return {static_cast<uintptr_t>(value.as_raw()),
-		        DebugFormatArgument::PermissionSet};
+		        DebugFormatArgumentKind::DebugFormatArgumentPermissionSet};
+	}
+};
+
+/**
+ * Null pointer specialisation.
+ */
+template<>
+struct DebugFormatArgumentAdaptor<std::nullptr_t>
```

Comment:
In the test, I wrote nullptr.  You would probably never do that in real code (why would you log a pointer that you know is null at compile time?) but it turned out that we were missing a specialisation for `std::nullptr_t`.  It does not automatically decay to `void*` so that you can have functions that have unambiguous overloads of `void*` and `std::nullptr` for use in `constexpr` things where you want to specialise on known-to-be-`nullptr`-at-compile-time.